DataFormats/MemoryResources/include/MemoryResources/MemoryResources.h

```diff
@@ -108,11 +108,7 @@ class ChannelResource : public FairMQMemoryResource
 
   bool do_is_equal(const boost::container::pmr::memory_resource& other) const noexcept override
   {
-    const FairMQMemoryResource* that = dynamic_cast<const FairMQMemoryResource*>(&other);
-    if (that && that->getTransportFactory() == factory) {
-      return true;
-    }
-    return false;
+    return this == &other;
```

Comment:
I have to say that the original code looked suspicious to me, see comment in https://github.com/AliceO2Group/AliceO2/pull/1140#discussion_r192346936. But the answer seamed to be reasonable. I still can not judge if that is the right thing now, but it looks less suspicious. How about the original intentions that "allocators are equal if resource allocated with one can be deallocatoed with another"?

Reply:
yes, your comment probably triggered some mental process leading to this fix :)
My original comment is still valid - BUT the allocator wrapper around the transport factory carries some unique state which is not available to the transport - so it needs to be the same allocator wrapper (the ChannelResource) that deallocates. Now if a transport factory would BE the allocator, it would be possible. I jumped ahead of myself a bit there... As I noted: a brown-paper-bag thing :)

Reply:
ok, thanks